main/models.py

```diff
@@ -838,3 +838,98 @@ def clean(self) -> None:
+    )
+
+    def __str__(self) -> str:
+        """String representation of the ProjectPhase object."""
+        return f"{self.project.name} - {self.start_date} -> {self.end_date}"
+
+    def save(self, **kwargs: Any) -> None:  # type: ignore
+        """Saves the object to the database.
+
+        This overwrites models.Model.save() to keep the days constant if the start or
+        end date changes, modifying the FTE value. Except if `value` has also changed in
+        the same modification.
+        """
+        update_fields = kwargs.get("update_fields")
+        # collects updated fields if there are any and checks for date changes
+        if update_fields is not None and (
+            "start_date" in update_fields or "end_date" in update_fields
+        ):
+            if "value" not in update_fields:  # if date change and no value change
+                # get old date (from DB)
+                old_days = ProjectPhase.objects.get(pk=self.pk).days
+                # update the value keeping the days constant by updating FTE value
+                self.value = old_days / (
+                    (self.end_date - self.start_date).days * WORKING_DAYS / 365
+                )
+                kwargs["update_fields"] = {"value"}.union(update_fields)
+
+        super().save(**kwargs)
+
+    def clean(self) -> None:
+        """Ensures that phase dates are sensible.
+
+        Ensures start is before the end date.
+        Ensures phase within project period.
+        Ensures the phase isn't covered by any other phases.
+        Ensures at least phase start or end date aligns with other phases or project
+            dates.
+        Ensures project has funding before phase added.
+        """
+        super().clean()
+        if (
+            self.project.start_date > self.start_date  # type: ignore [operator]
+            or self.project.end_date < self.end_date  # type: ignore [operator]
+        ):
+            raise ValidationError(
+                "Phase period must be within the project period: "
+                f"{self.project.start_date} -> {self.project.end_date}"
+            )
+
+        # Check for overlapping phases
+        overlapping = ProjectPhase.objects.filter(
+            project=self.project,
+            start_date__lt=self.end_date,  # Starts that are before the new end
+            end_date__gt=self.start_date,  # Ends that are after the new start
+        )
+
+        # Exclude self if this is an update (not a new instance)
+        if self.pk:
+            overlapping = overlapping.exclude(pk=self.pk)
+
+        if overlapping.exists():
+            first_conflict = overlapping.first()
+            raise ValidationError(
+                "Phase period must not overlap with other phase periods for the same "
+                f"project: {first_conflict.start_date} -> "  # type: ignore [union-attr]
+                f"{first_conflict.end_date} vs. {self.start_date} -> {self.end_date}"  # type: ignore [union-attr]
```

Comment:
Not sure why you need to ignore this. start and end dates are mandatory for FTEs, right?